File: problem30.rb
def sum_of_fifth_powers
  result = []
  2.upto(354_294) do |num|
  	new_num = num
    num.to_s.split("").each do |digit|
      new_num -= (digit.to_i) ** 5
           

    end  

    if new_num == 0
      result << num
    end

  end

  result.reduce(:+)

end

p sum_of_fifth_powers

File: problem30.py
def sum_of_fifth_powers():
  upper_limit = 5 * (9 ** 5)
  i = 2
  result = 0
  while (i < upper_limit):
    new_num = i
    num_string = str(i)
    string_idx = 0
    while (string_idx < len(num_string)):
      digit = int(num_string[string_idx])
      new_num -= digit ** 5
      string_idx += 1
    if (new_num == 0):
      result += i
    i += 1
  return result

print sum_of_fifth_powers()
